### PR TITLE
refactor(conference): add CoroutineScope and re-implement message()

### PR DESCRIPTION
**Pull request by drymarau**

### CODE REVIEW

File: sdk-conference-infinity/src/main/kotlin/com/pexip/sdk/conference/infinity/InfinityConference.kt [hunk 1]
```diff
@@ -43,6 +46,7 @@ public class InfinityConference private constructor(
 ) : Conference {
 
     private val executor = Executors.newSingleThreadScheduledExecutor()
+    private val scope = CoroutineScope(SupervisorJob() + executor.asCoroutineDispatcher())
```
Comment (tpettersen) [inline sdk-conference-infinity/src/main/kotlin/com/pexip/sdk/conference/infinity/InfinityConference.kt:49]: Just trying to understand the reason for keeping the executor and not just go for `Dispatchers.Default`. Any input?

Comment (drymarau) [reply sdk-conference-infinity/src/main/kotlin/com/pexip/sdk/conference/infinity/InfinityConference.kt:49]: Slow and steady migration 🙂

**Review by tpettersen — APPROVED** [review]
Works good :)

File: sdk-conference-infinity/src/main/kotlin/com/pexip/sdk/conference/infinity/InfinityConference.kt [hunk 2]
```diff
@@ -83,32 +87,25 @@ public class InfinityConference private constructor(
...
-                }
-            },
-        )
+        scope.launch {
+            val message = try {
+                messenger.send(type = "text/plain", payload = payload)
+            } catch (e: MessageNotSentException) {
+                return@launch
+            }
+            val event = MessageReceivedConferenceEvent(
```
[the remark this answers is not in the document]
Comment (drymarau) [reply sdk-conference-infinity/src/main/kotlin/com/pexip/sdk/conference/infinity/InfinityConference.kt:96]: It's a way to inject your sent message to the event stream so it's easier to implement chat and the like - Infinity itself doesn't echo it.

[the remark this answers is not in the document]
Comment (tpettersen) [reply sdk-conference-infinity/src/main/kotlin/com/pexip/sdk/conference/infinity/InfinityConference.kt:96]: yes, that I understood. Was more thinking semantically and naming wise :)

**Pull request merged**